app/subcommands/stop_subcommand.py

```diff
@@ -62,23 +62,33 @@ def _kill_pid_in_file_if_exists(self, pid_file_path):
 
         if not matched_proc_command:
             self._logger.info(
-                "PID {0} is running, but command '{1}' is not a clusterrunner command".format(pid, proc_command))
+                "PID {0} is running, but command '{1}' is not a clusterrunner command".format(proc.pid, proc_command))
             return
 
         # Try killing gracefully with SIGTERM first. Then give process some time to gracefully shutdown. If it
         # doesn't, perform a SIGKILL.
         # @TODO: use util.timeout functionality once it gets merged
-        os.kill(int(pid), signal.SIGTERM)
+        procs_to_kill = proc.get_children() + [proc]
+        self.kill_running_procs(procs_to_kill, signal.SIGTERM)
         sigterm_start = time.time()
 
-        while (time.time()-sigterm_start) <= self._sigterm_sigkill_grace_period_sec:
-            if not psutil.pid_exists(int(pid)):
+        while (time.time()-sigterm_start) <= self.SIGTERM_SIGKILL_GRACE_PERIOD_SEC:
+            if all([not ptk.is_running() for ptk in procs_to_kill]):
```

Comment:
`proc_to_kill` instead of `ptk`? If there's a perfect level of verbosity in naming, I'd rather be on the more verbose side. I think a little goes a long way in readability. :)